src/benchmarklib/tpch/tpch_query_generator.cpp

```diff
@@ -52,6 +70,336 @@ std::string TPCHQueryGenerator::get_preparation_queries() const {
 }
 
 std::string TPCHQueryGenerator::build_query(const QueryID query_id) {
+  using namespace std::string_literals;  // NOLINT
+
+  // Preferring a fast random engine over one with high-quality randomness. Engines are not thread-safe. Since we are
+  // fine with them not being synced across threads and object cost is not an issue, we simply use one generator per
+  // calling thread.
+  static thread_local std::minstd_rand random_engine{};
+
+  // This is not nice, but initializing this statically would require external methods and make it harder to
+  // follow in the end. It's not like this list will ever change...
+  static const std::vector materials{"TIN", "NICKEL", "BRASS", "STEEL", "COPPER"};
+
+  static const auto sizes =
+      std::vector{1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25,
+                  26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50};
+
+  static const auto country_codes =
+      std::vector{10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34};
+
+  // Random distributions for all strings defined by the TPC-H benchmark. Distributions should not be modified when
+  // they are used, but because we have no explicit thread safety guarantee, we make this thread-local, too.
+  static thread_local std::uniform_int_distribution<> material_dist{0, static_cast<int>(materials.size() - 1)};
+  static thread_local std::uniform_int_distribution<> region_dist{0, regions.count - 1};
+  static thread_local std::uniform_int_distribution<> segment_dist{0, c_mseg_set.count - 1};
+  static thread_local std::uniform_int_distribution<> nation_dist{0, nations.count - 1};
+  static thread_local std::uniform_int_distribution<> type_dist{0, p_types_set.count - 1};
+  static thread_local std::uniform_int_distribution<> color_dist{0, colors.count - 1};
+  static thread_local std::uniform_int_distribution<> shipmode_dist{0, l_smode_set.count - 1};
+  static thread_local std::uniform_int_distribution<> brand_char_dist{1, 5};
+  static thread_local std::uniform_int_distribution<> container_dist{0, p_cntr_set.count - 1};
+
+  // Will be filled with the parameters for this query and passed to the next method which builds the query string
+  std::vector<std::string> parameters;
+
+  switch (query_id) {
+    // Writing `1-1` to make people aware that this is zero-indexed while TPC-H query names are not
+    case 1 - 1: {
+      // Random values, such as 60-120 days, are taken from the TPC-H specification. Each query in Chapter 2.4 has a
```

Comment:
I'd put this above the `switch (query_id) {` as this applies to all queries, not just query 1